docs/advanced-usage.md

```diff
@@ -0,0 +1,145 @@
+# Advanced usage
+
+FakeItEasy exposes a few APIs that aren't commonly needed, but can be useful in certain scenarios.
+
+## Clearing the configuration of a fake
+
+The `Fake.ClearConfiguration` method removes all configured rules from a fake, leaving it in the default configuration.
+
+```csharp
+var foo = A.Fake<IFoo>();
+Assert.Equal(0, foo.Bar());
+A.CallTo(() => foo.Bar()).Returns(42);
+Assert.Equal(42, foo.Bar());
+Fake.ClearConfiguration(foo);
+Assert.Equal(0, foo.Bar());
+```
+
+## Clearing a fake's recorded calls
+
+The `Fake.ClearRecordedCalls` method clears all recorded calls from a fake. Subsequent call assertions won't see these calls.
+
+```csharp
+var foo = A.Fake<IFoo>();
+foo.Bar();
+A.CallTo(() => foo.Bar()).MustHaveHappened(); // Succeeds
+Fake.ClearRecordedCalls(foo);
+A.CallTo(() => foo.Bar()).MustNotHaveHappened(); // Succeeds
+```
+
+## Getting the list of calls made on a fake
+
+Sometimes, the call assertion API offered by FakeItEasy isn't enough, and you need to manually check the calls made on a fake.
+The `Fake.GetCalls` method returns a fake's recoreded calls as a list of `ICompletedFakeObjectCall` that you can examine yourself:
+
+```csharp
+var foo = A.Fake<IFoo>();
+foo.Bar();
+foo.Baz();
+var calls = Fake.GetCalls(foo).ToList();
+Assert.Equal(2, calls.Count);
+Assert.Equal("Bar", calls[0].Method.Name);
+Assert.Equal("Baz", calls[1].Method.Name);
+```
+
+## The `FakeManager` object
+
+The `Fake.GetFakeManager` method returns a `FakeManager` object that can be used to get information on the fake and manipulate its call rules.
+
+### Getting the type of the fake
+
+The `FakeManager.FakeObjectType` property returns the type of the fake, i.e. the type that was passed to `A.Fake`. This can be useful
+if you're writing code that dynamically manipulates fakes.
+
+```csharp
+var foo = A.Fake<IFoo>();
+var manager = Fake.GetFakeManager();
+Assert.Equal(typeof(IFoo), manager.FakeObjectType);
+```
+
+### Getting the fake from the fake manager
+
+The `FakeManager.Object` property returns the fake object managed by this `FakeManager`.
+
+```csharp
+var foo = A.Fake<IFoo>();
+var manager = Fake.GetFakeManager();
+Assert.Equal(foo, manager.Object);
+```
+
+### Manipulating a fake's call rules
+
+The `FakeManager.Rules` property returns all the rules configured on a fake.
+
+```csharp
+var foo = A.Fake<IFoo>();
+A.CallTo(() => foo.Bar()).Returns(42);
+A.CallTo(() => foo.Baz()).DoesNothing();
+var manager = Fake.GetFakeManager(foo);
+var rules = manager.Rules.ToList();
+Assert.Equal(2, rules.Count);
+```
+
+It is also possible to add custom rules for advanced scenarios, using `AddRuleFirst` and `AddRuleLast`. `AddRuleFirst` adds a rule at the beginning of the rule list, so that it's evaluated before any other rule. `AddRuleLast` adds a rule at the end of the rule list, so that it's evaluated after any other rule (which is the normal behavior when configuring a fake with  `A.CallTo(...)`).

```

Comment:
>     1. I worry that "evaluated after any other rule" connotes that the rules are all executed in order, instead of each rule being assessed for applicability and only the first applicable one being executed

Good point. Any suggestion on how to express this?

>     2. fun fact: the normal behaviour when configuring with `A.CallTo(…)` is equivalent to `AddRuleFirst`. New behaviour overrides old behaviour!

Oops. Not sure what I was thinking.

Reply:
> > 1. I worry that "evaluated after any other rule" connotes that the rules are all executed in order, instead of each rule being assessed for applicability and only the first applicable one being executed

> Good point. Any suggestion on how to express this?

I think switching "evaluate before" to "considered before" might be sufficient. But maybe that's just my ear…

Reply:
I was thinking something similar. Thanks!